tensorflow/compiler/xla/service/gpu/BUILD

```diff
@@ -1357,19 +1356,25 @@ tf_cc_test(
 
 cc_library(
     name = "buffer_comparator",
-    srcs = ["buffer_comparator.cc"],
+    srcs = [] + if_cuda_is_configured([
```

Comment:
Part of the discussion addressed in [above comments](https://github.com/tensorflow/tensorflow/pull/30884#discussion_r315300835). What I can do is to bring the `ROCm` stream executor side interface completely to parallel (to create a dummy interface but return `unimplemented error` for everything), so that both `Cuda` and `ROCm` implementation can be compiled and included in the same target. Then it is up to the runtime to choose the correct implementation based on platform id on the fly. 

Regarding the suggestion, I thought `#ifdef` is something we want to avoid in the code base? @timshen91 Just for clarification.

Reply:
This is perhaps a most important point for this and similar CLs, so it's definitely worth discussing.

In terms of maintainability and complexity, surrounding everything with `#ifdef` is worse than switching source at build time, which is worse than an isolated `#ifdef`, which is worse than runtime dispatch (either virtual or an explicit switch).

I would prefer to avoid build-time tricks altogether, since they make the code less maintainable: it becomes not possible to build a binary capable of doing both, standard tooling stops working (since now #ifdef values have to be passed), getting build errors gets harder (as two build commands are required), etc etc.